pep-0681.rst

```diff
@@ -714,6 +668,26 @@ We chose not to support this feature, since it is specific to
 SQLAlchemy. Users can manually set ``default=None`` on these fields
 instead.
 
+Descriptor-typed field support
+------------------------------
+
+We considered adding a boolean parameter on ``dataclass_transform``
+to enable better support for fields with descriptor types, which is
+common in SQLAlchemy. When enabled, the type of each parameter on the
+synthesized ``__init__`` method corresponding to a descriptor-typed
+field would be the type of the value parameter to the descriptor's
+``__set__`` method rather than the descriptor type itself. Similarly,
+when setting the field, the ``__set__`` value type would be expected.
+And when getting the value of the field, its type would be expected to
+match the return type of ``__get__``.
+
+This idea was based on the belief that ``dataclass`` did not properly
+support descriptor-typed fields. In fact it does, but type checkers
+(at least mypy and pyright) did not reflect the runtime behavior which
+led to our misunderstanding. For more details, see the
+`Pyright bug <#pyright-descriptor-bug>`_.
```

Comment:
```suggestion
`Pyright bug <#pyright-descriptor-bug_>`__.
```

The `_` is required after the reference name for the link to work; it is currently broken as is.

I've developed and tested a linting check for this, since it seems to be a pretty easy mistake to make, that will catch this in the future.